Framework/PythonInterface/mantid/plots/datafunctions.py

```diff
@@ -584,6 +591,8 @@ def get_uneven_data(workspace, distribution):
     y = []
     nhist = workspace.getNumberHistograms()
     yvals = workspace.getAxis(1).extractValues()
+    if yvals == ['']:
```

Comment:
Can I ask what this is protecting against? The empty string is throwing me a little.

Reply:
If a workspace does not have spectra ID then `extractValues()` returns [''], which causes issues later

Reply:
I don't quite follow. Do you mean not having a `spectraAxis`?. `extractValues` should just pull out what values are on the vertical axis. Do you have an example that causes this?

Reply:
the output workspace from the origional issue has no spectra id for its spectra

![image](https://user-images.githubusercontent.com/49688535/88813290-a4f21800-d1b0-11ea-8610-3f9ebef59eae.png)

the first workspace here returns [''] from extractValues()

Reply:
Thanks for the example I understand the issue now. The vertical axis is a `TextAxis` that happens to have a single empty string as its value.
I think here we are dealing with the specific case of a single empty string entry but I can create an TextAxis and I get an error plotting that too so I think we need to expand and better deal with text axes in general.

Using `CreateWorkspace` we can set a text axis with some values:

```python
CreateWorkspace(OutputWorkspace='textaxis', DataX='1,2,3,1,2,3', DataY='2,2,2,2', NSpec=2, VerticalAxisUnit='Text', VerticalAxisValues='a,b')
```

Trying a colorfill on this produces an error:

```
Traceback (most recent call last):
  File "/home/mgigg/Code/mantidproject/mantid-nightly/qt/applications/workbench/workbench/plugins/workspacewidget.py", line 132, in _do_plot_colorfill
    pcolormesh(names)
  File "/home/mgigg/Code/mantidproject/mantid-nightly/Framework/PythonInterface/mantid/plots/plotfunctions.py", line 56, in inner_func
    return func(workspaces, *args, **kwargs)
  File "/home/mgigg/Code/mantidproject/mantid-nightly/qt/python/mantidqt/plotting/functions.py", line 201, in pcolormesh
    pcm = pcolormesh_on_axis(ax, ws)
  File "/home/mgigg/Code/mantidproject/mantid-nightly/qt/python/mantidqt/plotting/functions.py", line 250, in pcolormesh_on_axis
    if use_imshow(ws):
  File "/home/mgigg/Code/mantidproject/mantid-nightly/qt/python/mantidqt/plotting/functions.py", line 169, in use_imshow
    difference = np.diff(y)
  File "/usr/lib/python3/dist-packages/numpy/lib/function_base.py", line 1931, in diff
    return a[slice1]-a[slice2]
TypeError: ufunc 'subtract' did not contain a loop with signature matching types dtype('<U1') dtype('<U1') dtype('<U1')
```

I guess we need to deal with `TextAxis` differently. [`getAxis(1).isTextAxis()`](https://docs.mantidproject.org/nightly/api/python/mantid/api/TextAxis.html#mantid.api.TextAxis.isText) will return True if that axis is text. Maybe in that case we just use the indices [0, nhist) as the Y values?